src/test/java/AlinaTinca/Tests/Tema6/HerokuForms.java

```diff
@@ -0,0 +1,197 @@
+
+    //Test4
+        try{
+            navigateToHerokuIndexPage();
+            //go to the Heroku Index Page and identify the exception for goBackToTheForm button
+        }catch (NoSuchElementException e) {
+            FileUtils.takeScreenshot(driver, "new alert");
+            //call this method and take the screenshot for exception with the above message
+            System.out.println("It should take a screenshot for the exception identified");
+            //display the above message
+        }finally {
+            closeBrowser();
+            //close the driver instance
+        }
+    }
+
+    public static void navigateToHerokuFormsPage(){
+        driver= BrowserManager.createChromeDriver();
+        //create the driver instance
+        driver.get("https://testpages.herokuapp.com/styled/basic-html-form-test.html");
+        //go to the above URL
+        System.out.println("Heroku HTML Form Example page has been opened!");
+        //display the above message
+    }
+
+    public static void navigateToHerokuIndexPage(){
+        driver= BrowserManager.createChromeDriver();
+        //create the driver instance
+        driver.get("https://testpages.herokuapp.com/styled/index.html");
+        //go to the Heroku Index Page
+        WebElement goBackToTheFormButton = driver.findElement(By.className("styled-click-button"));
+        //find the goBackToTheForm button
+        goBackToTheFormButton.click();
+        //click on this button
+    }
+
+    public static void firstParagraph() {
+        WebElement firstParagraph = driver.findElement(By.className("explanation"));
+        //find the first paragraph
+        System.out.println("The first paragraph has the following message: " + firstParagraph.getText());
+        //display the message for the first paragraph
+    }
+
+    public static void scrollToSubmitButton() {
+        WebElement submitButton = driver.findElement(By
+                    .xpath("//input[@type='submit']"));
+        //find the submit button
+        Actions actions = new Actions(driver);
+        //create an object of type Actions
+        actions.moveToElement(submitButton).build().perform();
+        //scroll to the submit button
+        System.out.println("Scroll to the submit button");
+        //display the above message
+    }
+
+    public static void fillFormDetails(){
+        WebElement userName = driver.findElement(By.cssSelector("input[name='username']"));
+        //find the Username field
+        WebElement password = driver.findElement(By.cssSelector("input[name='password']"));
+        //find the Password field
+        WebElement textAreaComment = driver.findElement(By.cssSelector("textarea[name='comments']"));
+        //find the TextArea Comment field
+        WebElement fileUpload = driver.findElement(By.cssSelector("input[name='filename']"));
+        //find the Filename field
+        List <WebElement> checkBoxes = (List<WebElement>) driver.findElement(By.name("checkboxes[]"));
```

Comment:
In case you want tot find all checkboxes, use driver.findElements and not driver.findElement